### DISCUSSION TITLE
How to filter results based on multiple relational field values if available for an m-to-n relationship?

### DISCUSSION THREAD

**Discussion by palashCItobuz** [prisma/prisma#10534] [Q&A] [answered]
A company can have multiple industries and industry can have multiple companies. How to use prisma query to return companies only having specific industries? 
```prisma
model Company {
  id                     Int                   @id @default(autoincrement())
  leadSourceId           Int?                  @map("lead_source_id")
  createdAt              DateTime              @default(now()) @map("created_at")
  updatedAt              DateTime              @updatedAt @map("updated_at")
  fiscalYear             Int                   @default(dbgenerated("(date_part('year'::text, CURRENT_TIMESTAMP))::integer")) @map("fiscal_year")
  leadSource             User?                 @relation("leadSourceFor", fields: [leadSourceId], references: [id])
  companyIndustries      CompanyOnIndustries[]

  @@map("companies")
}


model CompanyOnIndustries {
  id         Int        @id @default(autoincrement())
  industryId Int        @map("industry_id")
  industry   Industries @relation(fields: [industryId], references: [id])
  companyId  Int        @map("company_id")
  company    Company    @relation(fields: [companyId], references: [id])
  createdAt  DateTime   @default(now()) @map("created_at")

  @@unique([industryId, companyId])
  @@map("company_on_industries")
}
```
Finally I need only the companies filtered by the industries. Return only the companies that has specific industry set in the relational table.

**Comment by palashCItobuz** [comment]
I have been trying to use this where but it's showing type error...
```
where: {
        companyIndustries: {
          where: {
            industries: {
              in: industries
            }
          },
        },
      }
```

**Comment by austincrim** [comment, marked as answer]
Hey @palashCItobuz 👋, thanks for the question.

Could you try something like this:
```ts
await prisma.company.findMany({
    where: {
      companyIndustries: {
        every: {
          industry: {
            OR: [{ name: 'some industry' }, { name: 'a different industry' }]
          }
        }
      }
    }
})
```

This uses our [relation filter](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#every) API to return the companies where all of the related `companyIndustries` have an`industry` with at least one of the given names in the `OR` field.

I'm not sure what your `Industry` model looks like, I just used the `name` field as an example of how to filter.

Let me know if that works!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
